### DISCUSSION TITLE
How to hide the filter from filtermenu but still enable the functionality

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [filamentphp/filament#6847] [Help]
Hide the filter from here,
![image](https://github.com/filamentphp/filament/assets/26206253/491942ff-e39c-4549-bb84-c6dd91429d4c)

But still allow the query parameter & `$this->tableFilters['state']['value'] = $state;`.
![Image](https://github.com/filamentphp/filament/assets/26206253/74c2182d-4672-45c7-ad44-a6ad32bcc54b)

**Comment by pboivin** [comment]
One option is to use a custom filter form. You'll be able to use `->extraAttributes()` on the field and hide it with CSS:
https://filamentphp.com/docs/2.x/tables/filters#custom-filter-forms

Otherwise, maybe you could explore around the `mount()` method of your page class. Check the URL parameters and modify the table query.

**Reply by lucacastelnuovo** [reply]
Thank you for the suggestion. This code becomes quite messy and still leaves some weird whitespace.
